suite/tests/linux/thread.c

```diff
@@ -83,22 +83,28 @@ static struct timespec sleeptime;
 int
 main()
 {
-    sleeptime.tv_sec = 0;
-    sleeptime.tv_nsec = 10 * 1000 * 1000; /* 10ms */
-
-    child_exit = false;
-    child_done = false;
-    child = create_thread(run, NULL, &stack);
-    assert(child > -1);
-
-    /* waste some time */
-    nanosleep(&sleeptime, NULL);
-
-    child_exit = true;
-    /* we want deterministic printf ordering */
-    while (!child_done)
+    // We need to create multiple threads to verify correctness of reset,
+    // which requires more than one threads.
+    // XXX: Using -reset_at_thread_count 2 causes an ASSERT failure, so we
```

Comment:
What assert is this?  We should file an issue if it's a pre-existing thing.  Does this happen with -reset_at_nth_thread?

Reply:
Yes, it occurs with `-reset_at_nth_thread 2` too.

```
Internal Error: DynamoRIO debug check failure: Not implemented @dynamorio/core/unix/signal_linux_aarch64.c:53 (0)
```

I'm having a hard time getting a stack trace  :/. For some reason, gdb doesn't break at `d_r_internal_error` or `save_fpstate`, even though it recognises them when setting the breakpoints.

Anyway, here's a stack trace I constructed manually:

```
save_fpstate at core/unix/signal_linux_aarch64.c:53
thread_set_self_context at core/unix/signal.c:2684
thread_set_self_mcontext at core/unix/signal.c:2771
check_wait_at_safe_spot at core/synch.c:770
dispatch_enter_fcache at core/dispatch.c:436
```
Let me try getting a proper stack trace and I'll file an issue for this.

Reply:
> For some reason, gdb doesn't break at d_r_internal_error or save_fpstate, even though it recognises them when setting the breakpoints.

Maybe you're setting them in drrun rather than in the application?

Reply:
Further discussion in #4496